camel-core/src/main/java/org/apache/camel/util/StringHelper.java

```diff
@@ -672,5 +672,36 @@ public static String trimToNull(final String given) {
 
         return trimmed;
     }
+    
+    /**
+     * Checks if the src string contains what
+     *
+     * @param src  is the source string to be checked
+     * @param what is the string which will be looked up in the src argument 
+     * @return true/false
+     */
+    public static boolean containsIgnoreCase(String src, String what) {
+        final int length = what.length();
+        if (length == 0) {
+            return true; // Empty string is contained
+        }
+
+        final char firstLo = Character.toLowerCase(what.charAt(0));
+        final char firstUp = Character.toUpperCase(what.charAt(0));
+
+        for (int i = src.length() - length; i >= 0; i--) {
+            // Quick check before calling the more expensive regionMatches() method:
+            final char ch = src.charAt(i);
```

Comment:
Without quick check

# Run complete. Total time: 00:00:28

Benchmark                                             Mode  Cnt       Score       Error   Units
ContainsIgnoreCaseTest.benchmark                     thrpt    2      ≈ 10⁻⁵              ops/us
ContainsIgnoreCaseTest.benchmark                      avgt    2  250265,305               us/op
ContainsIgnoreCaseTest.benchmark                    sample   20  226072,986 ± 11368,864   us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.00    sample       207618,048               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.50    sample       223608,832               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.90    sample       247699,866               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.95    sample       248499,405               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.99    sample       248512,512               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.999   sample       248512,512               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.9999  sample       248512,512               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p1.00    sample       248512,512               us/op
ContainsIgnoreCaseTest.benchmark                        ss    2  525478,432               us/op


With quick check
# Run complete. Total time: 00:00:27

Benchmark                                             Mode  Cnt       Score       Error   Units
ContainsIgnoreCaseTest.benchmark                     thrpt    2      ≈ 10⁻⁵              ops/us
ContainsIgnoreCaseTest.benchmark                      avgt    2  250323,754               us/op
ContainsIgnoreCaseTest.benchmark                    sample   20  205455,360 ± 19499,901   us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.00    sample       180879,360               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.50    sample       198443,008               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.90    sample       248879,514               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.95    sample       258827,878               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.99    sample       259260,416               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.999   sample       259260,416               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p0.9999  sample       259260,416               us/op
ContainsIgnoreCaseTest.benchmark:benchmark·p1.00    sample       259260,416               us/op
ContainsIgnoreCaseTest.benchmark                        ss    2  254005,954               us/op


There is a slight difference.
I don't think jmh results are so accurate to measure such a single line. for now i'll prefer keeping it.